3rdparty/taucs/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+cmake_minimum_required(VERSION 3.3)
+
+if (MSVC)
+    set(CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "" FORCE)
+    set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /D_DEBUG")
+    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /W4")
+else()
+    set(CMAKE_C_FLAGS_DEBUG "-Wall -fPIC -ggdb -pipe")
+    set(CMAKE_C_FLAGS_RELEASE "-Wall -fPIC -O3 -g")
+endif()
+
+if(WIN32)
+    set(TAUCS_OSTYPE win32)
+else()
+    set(TAUCS_OSTYPE linux)
+endif()
+
+add_executable(taucs_configurator configurator/taucs_config.c)
+set(ENV_CMD ${CMAKE_COMMAND} -E env "OSTYPE=${TAUCS_OSTYPE}")
+set(TAUCS_CONFIG_DIR ${CMAKE_CURRENT_BINARY_DIR}/build/linux)
+set(TAUCS_CONFIG_BUILD ${TAUCS_CONFIG_DIR}/taucs_config_build.h)
+set(TAUCS_CONFIG_TESTS ${TAUCS_CONFIG_DIR}/taucs_config_tests.h)
+
+
+file(WRITE ${TAUCS_CONFIG_TESTS} "#define TAUCS_CORE\n#define TAUCS_BLAS_NOUNDERSCORE\n#include <cblas.h>\n")
```

Comment:
at least on Fedora, cblas.h is found under /usr/include/cblas/cblas.h, so this should be either "include <cblas/cblas.h>" or "-I /usr/include/cblas" added to CFLAGS somewhere else. Maybe the latter, since Fedora might be mangling the install path of various include files w.r.t. the original cmakefile shipping with lapack/cblas...

Reply:
Thanks for pointing this out. It seems like Ubuntu and Arch both install it straight into /usr/include; so the latter seems like a better option.

Due to various problems with TAUCS, I am currently investigating alternative solver options, which might remove the BLAS dependency altogether in any case.

Reply:
TAUCS has been replaced by Eigen3 now.